content/docs/2.15/scalers/splunk.md

```diff
@@ -0,0 +1,279 @@
++++
+title = "Splunk"
+availability = "v2.15+"
+maintainer = "Community"
+category = "Data & Storage"
+description = "Scale applications based on Splunk saved search results."
+go_file = "splunk_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `splunk` trigger that scales based on the result of a [saved search](https://docs.splunk.com/Documentation/Splunk/9.2.1/Search/Savingsearches).
+
+The trigger always requires the following information:
+
+```yaml
+triggers:
+  - type: splunk
+    metadata:
+      host: https://splunk.default.svc.cluster.local:8089
+      username: admin
+      targetValue: "1"
+      activationValue: "10"
+      savedSearchName: my-saved-search-name
+      valueField: count
+```
+
+**Parameter list:**
+
+- `username` - Splunk username authorized to access the search API.
```

Comment:
I think we should probably place the `username` next to the `password` in the trigger auth. I suppose it will only be used together and the username without a password is not gonna work, right?

CC @kedacore/keda-maintainers

Reply:
For auth, a password or API token can be supplied and a username is required for either. I kept username in trigger metadata since it's not technically a secret and is required for both methods of authentication.

FWIW, the username is required in the API route when making requests, this is another reason why its a required field in trigger metadata.

Reply:
okay, makese sense. let's keep it this way.

Reply:
I can change the scaler to always source username from trigger auth if that's preferred.

Reply:
Let's see what other think, thanks 🙏

Reply:
I do think Zbynek has a point though, do we at least support passing username as part of trigger authentication? I do not see it so would prefer to add it  there.

Otherwise, end-users need to change 2 places if they want to change identities nor cannot re-use it across scalers and have to duplicate it everywhere (although the secret can be shared which defeats purpose)

Reply:
Fair point. I updated https://github.com/kedacore/keda/pull/5905 and the docs here to source `username` from auth params

Reply:
@circa10a and to be clear, the scaler can't be used withouth `username` ie never without TriggerAuthentication, is that right? (it is not a problem, just a clarification we might want to add to the docs).

Reply:
Correct. The operator will also error stating that username is supplied as well. Where is the most applicable place in the docs to specify this requirement?

Reply:
I am followin the conversation and I would suggest here: https://github.com/kedacore/keda-docs/pull/1413#discussion_r1661395761

WDYT @zroubalik ?

Reply:
Agree, something around the lines, that user should be aware that the scaler always requires `username` set